lib/pushmi_pullyu/aip/fedora_fetcher.rb

```diff
@@ -1,65 +1,59 @@
-require 'pushmi_pullyu'
-require 'pushmi_pullyu/aip'
 require 'net/http'
-require 'fileutils'
 
 class PushmiPullyu::AIP::FedoraFetcher
 
+  class FedoraFetchError < StandardError; end
+
   RDF_FORMAT = 'text/rdf+n3'.freeze
 
   def initialize(noid)
     @noid = noid
   end
 
-  def base_path
-    PushmiPullyu.options[:fedora][:base_path]
-  end
-
   def object_url(url_extra = nil)
     url = "#{PushmiPullyu.options[:fedora][:url]}#{base_path}/#{pairtree}"
     url += url_extra if url_extra
     url
   end
 
-  def pairtree
-    "#{@noid[0..1]}/#{@noid[2..3]}/#{@noid[4..5]}/#{@noid[6..7]}/#{@noid}"
-  end
+  # Return true on success, raise an error otherwise
+  # (or use 'optional' to return false on 404)
+  def download_object(download_path, url_extra: nil,
+                      optional: false, is_rdf: false)
 
-  def download_object(download_path: nil, url_extra: nil,
-                      optional: false, rdf: false)
-    # Return true on success, raise an error otherwise
-    # (or use 'optional' to return false on 404)
-
-    url = object_url(url_extra)
-    uri = URI(url)
+    uri = URI(object_url(url_extra))
 
     request = Net::HTTP::Get.new(uri)
     request.basic_auth(PushmiPullyu.options[:fedora][:user],
                        PushmiPullyu.options[:fedora][:password])
 
-    request['Accept'] = RDF_FORMAT if rdf
+    request['Accept'] = RDF_FORMAT if is_rdf
 
     response = Net::HTTP.start(uri.hostname, uri.port) do |http|
       http.request(request)
     end
 
     if response.is_a?(Net::HTTPSuccess)
-      if download_path
```

Comment:
This is weird. If we have no download_path, this if statement should be at the very start of the method. Why go hit the API if we arent going to do anything with the results? Let's just make this argument manatory instead of optional. Looks like this was done for testing. But as a result, your not even testing what this method is even doing then in a real world setting.